cilium/cmd/cleanup.go

```diff
@@ -320,14 +328,34 @@ func confirmCleanup() bool {
 func removeCNI() error {
 	os.Remove(cniConfigV1)
 	os.Remove(cniConfigV2)
+	os.Remove(cniConfigV3)
 
-	err := os.Remove(cniConfigV3)
+	err := os.Remove(cniConfigV4)
 	if os.IsNotExist(err) {
 		return nil
 	}
 	return err
 }
 
+// revertCNIBackup removes the ".cilium_bak" suffix from all files in cniPath,
+// reverting them back to their original filenames.
+func revertCNIBackup() error {
+	return filepath.Walk(cniPath,
```

Comment:
Just a small note here, `filepath.Walk` walks all directories recursive, not only the "top-level" one. It is likely to not affect anything but just a FYI.